R/nse-inject.R

```diff
@@ -400,55 +440,62 @@ NULL
 #' as_label(letters)
 #'
 #' @export
-englue <- function(x) {
-  check_string(x)
+englue <- function(x,
+                   env = caller_env(),
+                   error_call = current_env(),
+                   error_arg = "x") {
+  check_string(x, arg = error_arg, call = error_call)
 
   if (!grepl("{{", x, fixed = TRUE)) {
-    abort(c(
-      "Must use `{{`.",
-      i = "Use `glue::glue()` for interpolation with `{`."
-    ))
+    abort(
+      c(
+        "Must use `{{`.",
+        i = "Use `glue::glue()` for interpolation with `{`."
+      ),
+      arg = error_arg,
+      call = error_call
+    )
   }
 
-  glue_embrace(
-    x,
-    env = caller_env(),
-    error_call = caller_env()
-  )
+  glue_embrace(x, env = env)
 }
 
-glue_embrace <- function(text,
-                         env = caller_env(),
-                         error_call = caller_env()) {
-  out <- glue_first_pass(
-    text,
-    env = env,
-    error_call = error_call
-  )
+glue_embrace <- function(text, env = caller_env()) {
+  out <- glue_first_pass(text, env = env)
   unstructure(glue::glue(out, .envir = env))
 }
 
-glue_first_pass <- function(text,
-                            env = caller_env(),
-                            error_call = caller_env()) {
+glue_first_pass <- function(text, env = caller_env()) {
   glue::glue(
     text,
     .open = "{{",
     .close = "}}",
-    .transformer = function(...) {
-      glue_first_pass_eval(..., error_call = error_call)
-    },
+    .transformer = function(...) glue_first_pass_eval(...),
     .envir = env
   )
 }
-glue_first_pass_eval <- function(text, env, error_call) {
+glue_first_pass_eval <- function(text, env) {
   text_expr <- parse_expr(text)
   defused_expr <- eval_bare(call2(enexpr, text_expr), env)
 
-  if (is_symbol(text_expr)) {
+  if (is_symbol(text_expr) && is_missing(defused_expr)) {
+    error_arg <- as_string(text_expr)
+    error_call <- NULL
+
+    # Find the relevant error frame. There are edge cases where this
```

Comment:
This is the only part im a little iffy on but it seems reasonable and the downsides seem slim